### ISSUE TITLE
CI Nightly Runs

### ISSUE THREAD

**Issue by kmpaul** [NCAR/notebook-gallery#52]
We'd like to have the CircleCI tests run nightly, which involves adding a `workflows` section in the `.circleci/config.yml`:

```yaml
workflows:
  nightly:
    triggers:
      - schedule:
          cron: "0 0 * * *"
          filters:
            branches:
              only:
                - master
```

See `intake-esm` for an example.  Also want to add jobs for `upsteam-dev`, which involves pointing critical packages (e.g., xarray, dask, etc.) to the master branch of their GitHub repos.

**Comment by kmpaul** [comment]
Probably want the `upstream-dev` job to point to upstream master branches of xarray, dask, distributed, and matplotlib, but we can change this in the future.

**Comment by andersy005** [comment]
Fixed in #55
